index.js

```diff
@@ -1,88 +1,129 @@
 // ***** Question 1 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
+let counter = 1
 
-// console.log(counter) 
-// // => 2
-// console.log("%c----------", "color: red") 
+console.log("%cQuestion 1", "color: red") 
+counter = 2
+console.log(counter) 
+// => 2
+console.log("%c----------", "color: red") 
 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
-// *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
+// // *** otherwise, the error will stop the rest of your code from running
+const name = 'James'
+console.log("%cQuestion 2", "color: red") 
 
-// console.log(name)
-// => "Raffy"
+console.log(name)
+// // => "Raffy"
 // name = "Not Raffy" 
-// => TypeError
-// console.log("%c----------", "color: red") 
+// // => TypeError
+console.log("%c----------", "color: red") 
 
 
 
-// ***** Question 3 *****
+// // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red") 
 
-// console.log(drinkWater(12))
-// // => "Man I sure am thirsty"
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
+function drinkWater(current_thirst_level){
+      console.log("Man I sure am thirsty")
+      current_thirst_level -= 1;
+      console.log("Ahh that hits the spot")
+      return current_thirst_level;
+}
+console.log(drinkWater(12))
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
-
+console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum){
+    if (num === maybeNum){
+      return "same same"
+    } else if(num == parseInt(maybeNum)){ //don't understand this "after converting maybeNum to a number part", I am supposed to use parseInt
+       return "same same (but different)"
+    } else{
+        return "different"
+    }
+  
+}
```

Comment:
- This is good. Since you're using `parseInt` to explicitly convert maybeNum into a number data type, you could avoid using `==` and use `===` instead:

`} else if (num === parseInt(maybeNum)) {`